.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,60 @@
+#
+# This workflow will run unit tests and attempt to publish package to NPM.
```

Comment:
Our existing CI already runs tests on PR and push. Imo this new workflow should either  extend existing one or ensure no redundant testing